### DISCUSSION TITLE
How to compile

### DISCUSSION THREAD

**Discussion by regenerous** [adriankumpf/tesla_auth#6] [Q&A] [answered]
First off thank you for all of the work you've done with Teslamate and Tesla Auth. Amazing!

I would like to compile your Tesla Auth code myself and then run it but alas I'm not knowledgeable enough to know how to carry out this task. I can use Mac, Windows, or RPi so whichever one of these platform is easiest will work. Would you be so kind as to let me know what I would need to do to compile your code and create the executable myself? Is it as simple as just downloading the three .rs files (auth, htime, and main) into a directory and then running rustc main.rs within that directory?

Thank you for your time and help.

UPDATE: I did some digging and learned some more about Rust (never worked with or heard of it before). I decided to install Rust on the same RPi that Teslamate is installed on.

Following the instructions here in the Rust docs (https://doc.rust-lang.org/book/ch01-01-installation.html) I installed Rust and then Git, and then cloned this repo and ran cargo build. Let's see if it works.

UPDATE 2: Got the follower errors and build failed:
![Capture](https://user-images.githubusercontent.com/6217175/138509872-dc85dac0-108e-4d8b-9b9a-98160d297739.JPG)

Not sure how to proceed from here.

**Comment by regenerous** [comment, marked as answer]
Seems like I'm missing some dependencies but as I was trying to figure that out I realized that the Teslamate RPi install has no Desktop or GUI so running tesla_auth will not work (correct me if I'm wrong).

Therefore, I decided to build/compile tesla_auth on my Macbook and there it seems to work fine. 

For completeness sake the following are the steps I took to get it working on my Macbook:

1. Installed rust: `curl --proto '=https' --tlsv1.2 https://sh.rustup.rs -sSf | sh`
2. Cloned this repo: `git clone https://github.com/adriankumpf/tesla_auth.git`
3. Went into the new tesla_auth directory: `cd tesla_auth`
4. Ran: `cargo build`
5. This will create a new "target" directory. CD into it: `cd target`
6. CD into debug directory: `cd debug`
7. Run tesla_auth: `./tesla_auth` (don't forget the period in the front)
8. A new window should have opened with text fields for you to input your credentials.

**Comment by adriankumpf** [comment]
Glad you figured it out! Building on macOS is probably the easiest. On Linux WebKitGTK is needed, e.g. for Debian based distributions the following packages have to be installed::`webkit2gtk-4.0 libgtksourceview-3.0-dev libappindicator3-dev`
